services/QuillLMS/app/models/subscription.rb

```diff
@@ -352,4 +342,33 @@ def self.create_with_school_or_user_join school_or_user_id, type, attributes
     subscription
   end
 
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def self.stripe_purchase_completed?(checkout_session_id)
+    return false if checkout_session_id.nil?
+
+    # TODO: remove this blocking waits for webhooks to complete
+    sleep ENV.fetch('STRIPE_WAIT_FOR_WEBHOOK', 0).to_i
```

Comment:
Oh, I see, this is handling my concern about the webhooks completing after page load?

Reply:
Definitely a hack 😜.  I think pusher will be a more sensible approach.